lib/mix/tasks/compile.make.ex

```diff
@@ -218,4 +219,34 @@ defmodule Mix.Tasks.Compile.ElixirMake do
 
     Mix.shell().info("Compiling with make: #{exec} #{args}")
   end
+
+  # Returns a map of default environment variables
+  # Defauts may be overwritten.
+  defp default_env(config, env) do
+    root_dir = :code.root_dir()
+    erl_interface_dir = Path.join(root_dir, "usr")
+    [erts_dir] = Path.wildcard(Path.join(root_dir, "erts*"))
+
+    Map.merge(
+      %{
+        "MIX_TARGET" => to_string(Mix.target()),
```

Comment:
For both Mix.target you can read the env var directly or set a default: `System.get_env("MIX_TARGET") || "host"`.